lib/algos/list/tools.js

```diff
@@ -31,11 +32,49 @@ function inc(str) {
             String.fromCharCode(str.charCodeAt(str.length - 1) + 1)) : str;
 }
 
+/*
+ * Compares two Strings, s1 and s2, using UTF-8 lexicographic ordering.
+ * @function
+ * @param {String} s1 the first string to compare
+ * @param {String} s2 the second string to compare
+ * @return {number} -1, 0, or 1 if s1 is less than, equal or greater than s2 respectively
+ */
+function utf8Compare(s1, s2) {
+    const l1 = s1.length;
+    const l2 = s2.length;
+    const l = Math.min(l1, l2)
+
+    for (let i = 0; i < l; i++) {
+        let cp1 = s1.codePointAt(i);
+        let cp2 = s2.codePointAt(i);
+
+        if (cp1 < cp2) {
+            return -1;
+        } else if (cp1 > cp2) {
+            return 1;
+        }
+
+        if (cp1 > 0xFFFF) {
+            i++;
+        }
+    }
+
+    if (l1 < l2) {
```

Comment:
This if, else if, else block can just be `return Math.sign(l1 - l2);` ( I am pretty sure anyway )
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sign